src/Mutator/Arithmetic/EqualOrIdenticalAssignment.php

```diff
@@ -35,16 +35,36 @@
 
 namespace Infection\Mutator\Arithmetic;
 
+use Infection\Mutator\Classification;
+use Infection\Mutator\Definition;
 use Infection\Mutator\Util\Mutator;
 use PhpParser\Node;
 
 /**
  * @internal
  */
-final class AssignmentEqual extends Mutator
+final class EqualOrIdenticalAssignment extends Mutator
 {
+    public static function getDefinition(): ?Definition
```

Comment:
@infection/core what do you think of this idea of "definition"?

The thought process here is that I would like to:

- review the mutators (see if they make sense, see what is missing, of check if there is any cosmetic change to do)
- make sure there is a meaningful and friendly explanation of what the mutation operator do
- classify the mutator
- provide a "remedy": a solution or guideline on how to kill the mutant

In order to do so I'm proposing to provide a way for the mutators to declare what they are about (via `Definition` and `Classification`) and have a quality check test (in the autoreview `MutatorTest`) to warn about the mutators which don't have a definition.

The classification chosen so far are the "classic" ones taken from other mutation testing tools, the terminology being mostly pushed by mbj from mutant (ruby mutation testing tool). It obviously can be expanded later on or we can add more classification means, but I think the above is a good first base.

Since this push for more self-documented mutators, this also opens the door to a way to generate the mutators doc from the source code rather than requiring us to manually keep things in sync.

Reply:
Oopsy looks like I completely messed up my merges so the above description is not on the right mutator. Still showcase well the example though.

If you agree on the idea I'll submit a dedicated PR with just the introduction of the definition thing and in another PR (this one for example) add it to the mutators